File: cpp/n434_WordNum.cpp
// https://leetcode-cn.com/problems/number-of-segments-in-a-string/
#include <iostream>
#include <vector>
#include <string>
#include <set>
#include <algorithm>
using namespace std;

class Solution{
public:
    int countSegments(string s){
        if (s == "") return 0;
        int res = 0;
        for (size_t i = 0; i < s.size(); ++i){
            if ((i == 0 || s[i-1] == ' ') && s[i] != ' ')
                res += 1;
        }
        return res;
    }
};

int main(){
    return 0;
}

File: python/n434_WordNum.py
# https://leetcode-cn.com/problems/number-of-segments-in-a-string/
class Solution:
    def countSegments(self, s: str) -> int:
        if not s: return 0
        tmp = s + " "
        res = 0
        for i in range(0, len(tmp)-1):
            if tmp[i] != ' ' and tmp[i+1] == ' ':
                res += 1
        return res
        
if __name__ == "__main__":
    pass        